Programs/python.c

```diff
@@ -70,7 +103,11 @@ main(int argc, char **argv)
 
     /* Force again malloc() allocator to release memory blocks allocated
        before Py_Main() */
+#ifdef Py_DEBUG
```

Comment:
You're right that it's technically unrelated now, but an earlier iteration of the patch failed in debug mode without it - that iteration was using `Py_SetStandardStreamEncoding`, and when you do that, you allocate some memory that gets freed during interpreter startup *after* the default allocator has been configured. Without this change, that free attempt failed in debug mode due to the allocator mismatch.

So these guards mean that the "safe to call before `Py_Initialize`" APIs actually are safe to call from the CLI entry point.